src/rdpc/variantCallingAnalysesProcessor.ts

```diff
@@ -0,0 +1,224 @@
+  donors: StringMap<SangerAndMutectInfo>
+): DonorInfoMap => {
+  const result: DonorInfoMap = {};
+
+  Object.entries(donors).forEach(([donorId, info]) => {
+    initializeRdpcInfo(result, donorId);
+
+    if (info.sangerVC[0]) {
+      result[donorId].sangerVcsFirstPublishedDate = new Date(
+        Number(info.sangerVC[0].firstPublishedAt)
+      );
+    }
+
+    if (info.mutect[0]) {
+      result[donorId].mutectFirstPublishedDate = new Date(
+        Number(info.mutect[0].firstPublishedAt)
+      );
+    }
+  });
+
+  return result;
+};
+
+// find the earliest dates by comparing sanger/mutect analysis's first published dates
+export const getEarliestDateForDonor = (
+  donors: StringMap<SangerAndMutectInfo>
+): StringMap<SangerAndMutectInfo> => {
+  const result: StringMap<SangerAndMutectInfo> = {};
+  Object.entries(donors).forEach(([donorId, info]) => {
+    const wfInfo: SangerAndMutectInfo = { sangerVC: [], mutect: [] };
+    const earliestSanger = _.head(
+      _.sortBy(info.sangerVC, (sanger) => Number(sanger.firstPublishedAt))
+    );
+    if (earliestSanger) {
+      wfInfo.sangerVC.push(earliestSanger);
+    }
+
+    const earliestMutect = _.head(
+      _.sortBy(info.mutect, (mutect) => Number(mutect.firstPublishedAt))
+    );
+    if (earliestMutect) {
+      wfInfo.mutect.push(earliestMutect);
+    }
+    result[donorId] = wfInfo;
+  });
+  return result;
+};
+
+export const getAllMergedDonor_variantCalling = async ({
+  analysesFetcher = fetchVariantCallingAnalyses,
+  egoJwtManager,
+  studyId,
+  url,
+  config,
+  donorIds,
+}: {
+  studyId: string;
+  url: string;
+  egoJwtManager: EgoJwtManager;
+  donorIds?: string[];
+  config: {
+    chunkSize: number;
+    state?: StreamState;
+  };
+  analysesFetcher: typeof fetchVariantCallingAnalyses;
+}): Promise<StringMap<SangerAndMutectInfo>> => {
+  const mergedDonors: StringMap<SangerAndMutectInfo> = {};
+
+  if (donorIds) {
+    for (const donorId of donorIds) {
+      logger.info(`streaming analyses for donor ${donorId}`);
+      const stream = variantCallingStream({
+        studyId,
+        rdpcUrl: url,
+        egoJwtManager,
+        config,
+        analysesFetcher,
+        donorId,
+      });
+      for await (const page of stream) {
+        if (page.length === 0) {
+          logger.info(`No variant calling analyses for streaming`);
+        }
+        logger.info(`Streaming ${page.length} of variant calling analyses...`);
+
+        const donorPerPage = convertAnalysis(page);
+        mergeAllDonors(mergedDonors, donorPerPage);
+      }
+    }
+  } else {
+    const stream = variantCallingStream({
+      studyId,
+      rdpcUrl: url,
+      egoJwtManager,
+      config,
+      analysesFetcher,
+    });
+    for await (const page of stream) {
+      if (page.length === 0) {
+        logger.info(`No variant calling analyses for streaming`);
+      }
+      logger.info(`Streaming ${page.length} of variant calling analyses...`);
```

Comment:
The wording in this log statement is unclear, can it be rephrased to be easier to understand?